infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesDeployments.java

```diff
@@ -125,26 +125,43 @@ public Pod deploy(Pod pod) throws InfrastructureException {
     ObjectMeta metadata = pod.getMetadata();
     PodSpec podSpec = pod.getSpec();
     podSpec.setRestartPolicy("Always"); // Only allowable value
+    final CompletableFuture<Pod> createFuture = new CompletableFuture<>();
+    final Watch createWatch =
+        clientFactory
+            .create(workspaceId)
+            .pods()
+            .inNamespace(namespace)
+            .watch(new CreateWatcher(createFuture, workspaceId));
     try {
-      Deployment deployment =
-          clientFactory
-              .create(workspaceId)
-              .extensions()
-              .deployments()
-              .inNamespace(namespace)
-              .createNew()
-              .withMetadata(metadata)
-              .withNewSpec()
-              .withReplicas(1)
-              .withNewTemplate()
-              .withMetadata(metadata)
-              .withSpec(podSpec)
-              .endTemplate()
-              .endSpec()
-              .done();
-      return awaitAndGetPod(deployment);
+      clientFactory
+          .create(workspaceId)
+          .extensions()
+          .deployments()
+          .inNamespace(namespace)
+          .createNew()
+          .withMetadata(metadata)
+          .withNewSpec()
+          .withReplicas(1)
+          .withNewTemplate()
+          .withMetadata(metadata)
+          .withSpec(podSpec)
+          .endTemplate()
+          .endSpec()
+          .done();
+      return createFuture.get(POD_CREATION_TIMEOUT_MIN, TimeUnit.MINUTES);
     } catch (KubernetesClientException e) {
       throw new KubernetesInfrastructureException(e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new InfrastructureException(
```

Comment:
I see that the same exception is thrown in other components with similar logic like `KubernetesNamespace`, `KubernetesIngresses` but `RuntimeStartInterruptedException` may be better here. 
I would say that it is just a note since it is not related to this PR. I'll try to take a look how it may influence workspace start interruption and I will create an issue if needed.

Reply:
It may be useful to have an identifier of pod or deployment in error message.

Reply:
I've added identifier (pod name) to the errors thrown here. 

Regarding the `RuntimeStartInterruptedException`, I'm not sure what we should use for env or ownerId here. Also, an interrupted exception is handled in [`KubernetesInternalRuntime`](https://github.com/eclipse/che/blob/44588aa775e77f629eb0a0052ac20d949ab8c7c3/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java#L214)